2333/2/libs/rapper.rb

```diff
@@ -0,0 +1,34 @@
+# class Rapper
+class Rapper
+  attr_reader :name, :battles, :number_of_bad_words
+  ROUNDS_ON_BATTLE = 3
+
+  def initialize(name, battles)
+    @name = name
+    @battles = battles
+    @number_of_bad_words ||= bad_words
```

Comment:
You can move this memoization variable `@number_of_bad_words` right to the `#bad_words` method.